python/tvm/relax/transform/legalize_ops/statistical.py

```diff
@@ -68,13 +76,23 @@ def te_std(x: te.Tensor, axis: List[tir.IntImm], keepdims: bool) -> te.Tensor:
 
 @register_legalize("relax.variance")
 def _variance(bb: BlockBuilder, call: Call) -> Expr:
-    return bb.call_te(
-        _te_variance,
-        call.args[0],
-        call.attrs.axis,
-        call.attrs.keepdims,
-        primfunc_name_hint="variance",
+    # Reductions often overflow fp16 values. If we encounter one, just cast to fp32.
+    data = call.args[0]
+    original_dtype = call.args[0].struct_info.dtype
+    if original_dtype == "float16":
+        data = bb.normalize(bb.emit_te(topi.cast, data, "float32"))
```

Comment:
We may need the same casting for Gelu because topi.erf doesn't support fp16.

Reply:
That case is a little trickier because gpus do support fp16 for erf, its only cpu that cant. I dont see target handling in legalization yet.